### PR TITLE
Add Dockerfile-dev

### PR DESCRIPTION
**Pull request by tsiegleauq**
Adds the Dockerfile-dev which executes the developing starting commands
in a docker container.
Required to attach other services using docker environment.

Adds the docker building instructions to the makefile

### CODE REVIEW

File: Dockerfile-dev [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM python:3.8
+
+RUN apt-get -y update && \
```
Comment (FinnStutzenstein) [inline Dockerfile-dev]: if `wait-for-it` not used in an entrypoint script, this RUN-command can be removed.

**Comment by normanjaeckel** [comment]
How can I develop inside a docker container (and commit changes and so on)? Maybe we have to map the code into a volume that is the dev repo outside.

File: Dockerfile-dev (hunk 1 above)
Comment (tsiegleauq) [reply Dockerfile-dev]: you still need the update and upgrade, no?

Comment (FinnStutzenstein) [reply Dockerfile-dev]: Bo, the image provides all needed (up-to-date) software. The update is necessary to fetch the apt-repositories, so apt can find wait-for-it. I think upgrade wasn't necessary at all. But maybe you ask Gernot for a confirmation on this.

**Comment by tsiegleauq** [comment]
> How can I develop inside a docker container (and commit changes and so on)? Maybe we have to map the code into a volume that is the dev repo outside.

Edited the `docker-run-dev` command to:
`docker run -it -v ${PWD}/openslides_backend:/srv/code/openslides_backend -p 8000:8000 --rm os4-backend`

Running `docker-run-dev` now starts a (not silent) server that refreshes once you make changes inside the "openslides_backend"

File: Dockerfile-dev (hunk 1 above)
Comment (tsiegleauq) [reply Dockerfile-dev]: @gsiv ?

Comment (tsiegleauq) [reply Dockerfile-dev]: Gernot and I removed the commando and changed to image back to alpine

File: Dockerfile-dev [hunk 2]
```diff
@@ -0,0 +1,17 @@
+FROM python:3.8-alpine
+
+RUN apk add build-base
```
Comment (normanjaeckel) [inline Dockerfile-dev]: I think even this run command is not necessary.

Comment (tsiegleauq) [reply Dockerfile-dev]: pip will fail without GCC, which alpine does not bring. It seems that Build-base the recommended way to get GCC.

https://wiki.alpinelinux.org/wiki/GCC

**Comment by tsiegleauq** [comment]
- Rebased on current master
- Adjusted the Docker-dev from using `start.py` to gunicorn, which seems to reload way faster

**Comment by normanjaeckel** [comment]
Sorry for the new merge conflict. I want to play a little bit with docker and developing inside it and then take your commit. You don't have to care about at the moment. Thanks.

**Comment by normanjaeckel** [comment]
Rebased and changed in #15

**Comment by tsiegleauq** [comment]
could you explain what are you doing?

**Comment by normanjaeckel** [comment]
I rebase your commit and added another. Because I did not want to force push into your branch I just opened a new pull request.

**Pull request closed without merging**